aicssegmentation/structure_wrapper/WorkflowEngine.py

```diff
@@ -192,3 +197,11 @@ def get_all_params(self) -> List[Dict[str, Any]]:
         for step in self.steps:
             all_params.append(step.get_params())
         return all_params
+
+    def get_widget_data(self, step: WorkflowStep) -> dict[str, Any]:
+        for k, v in self.widget_info.items():
```

Comment:
Instead of looping through the entire all_functions.json and finding things by module, can you just look it up with the function name like `self.widget_info[step.name]`? Or are names not reliable enough?

Reply:
Names differ between `all_functions.json`, and the individual workflow configs. I found that they just arent that reliable to use to search `all_functions.json` so I'm doing this search with module and function names.

I think its fine since its only slightly slower than doing `self.widget_info[step.name]` to get the information.

Alternatively we could have jianxu fix the naming between all_functions and all the individual workflow configs.

Reply:
Also im no longer including this code in WorkflowEngine but rather WorkflowStep since each step needs information about its widget individually.